### ISSUE TITLE
Automatically deploy release candidates to a sync test staging environment

### ISSUE THREAD

**Issue by thepiwo** [aeternity/ae_mdw#816]
1. any release branch should be automatically deployed and synced, there a backchannel for errors or exposure of logs would be helpful, if a new commit is pushed then the sync should probably be restarted so we can find issues early (TBD)
2. any tagged release should be sync tested from 0 to top  a backchannel for errors or exposure of logs as well as creation of a db snapshot once finished would be great (TBD)

Anything else that would help you or we have already discussed @venimus @jyeshe @sborrazas

**Comment by jyeshe** [comment]
If possible to make the sync restart optional after a commit it would be great for minor or endpoint fixes. For the tagged versions I liked the idea. It would nice to have access to a grafana dashboard like yours (: applied to this new staging env. A final request would be to have a way to generate snaphots on demand so that we can go quickly to a certain height state synced with a specific version.

**Comment by sborrazas** [comment]
I agree with @jyeshe, not all releases are going to need it. We could use perhaps github tags to specify whether we want to run it or not.

**Comment by venimus** [comment]
I think the major issue with backups and resync is that these require a lot of resources I tried with some smaller (cheaper) instances before but sync takes forever (like weeks). So if we want to maintain multiple releases in parallel that could be a problem. Also snapshots are huge and it would take hours to make and upload, meanwhile the snapshotted node should be down, meaning we can not simply use the main ones unattended. We can have a staging nodes that syncs and be deployed with latest master and one with latest release though.

In addition not really sure how you will trigger the sync at specific height and if you want to do snapshot on certain period, this mean the node will stop syncing every time for some time (becoming hours at certain point) then these snapshots would quickly escalate in size, so we need to upload as soon as they are done and deleted so the instance storage not getting full.

Tags seem ok at first glance, but still you will need to create a unique tag each time, which quickly becomes annoying and also often prone to human errors and conflicting situations (like pushing tags too fast). We still have limited machines and unlimited tags, so that will be really messy. 

I suggest to start really simple

**Comment by sborrazas** [comment]
@venimus I agree that implementing this workflow might be complicated indeed. Perhaps we can start incrementally, like you said, really simple. 

How would you suggest we should start out then?
